### PR TITLE
YSHOP2-1216: Added DropdownMenu component

### PR DESCRIPTION
**Pull request by GhitaNektt**
## JIRA Ticket

[YSHOP2-1216](https://youcanshop.atlassian.net/browse/YSHOP2-1216)

## QA Steps

* [ ] Go to http://localhost:3000/
* [ ] Check the Dropdown menu

## Note

Leave empty when you have nothing to say

[YSHOP2-1216]: https://youcanshop.atlassian.net/browse/YSHOP2-1216?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: packages/vue3/src/components/DropdownMenu/DropdownMenu.vue [hunk 1]
```diff
@@ -0,0 +1,32 @@
+<script setup lang="ts">
+import { ref } from 'vue';
+import { onClickOutside } from '@vueuse/core';
+
+const show = ref(false);
+const DropdownMenu = ref();
+const toggleMenu = () => {
```
Comment (eihabkhan1) [inline packages/vue3/src/components/DropdownMenu/DropdownMenu.vue:7]: Always separate refs from others

File: packages/vue3/src/components/DropdownMenu/DropdownMenu.vue [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+
+  nextTick(() => {
+    if (DropdownMenuTrigger.value && DropdownMenuContent.value && show.value) {
+      const { top, left } = setPosition(DropdownMenuContent.value, DropdownMenuTrigger.value, props.position, 5);
+      DropdownMenuContent.value?.setAttribute('style', `top:${top}px;left:${left}px`);
+    }
+  });
+};
+
+onClickOutside(DropdownMenuContainer, () => show.value = false);
```
Comment (mosarabi) [inline packages/vue3/src/components/DropdownMenu/DropdownMenu.vue:25]:
```suggestion
const handleScroll = () => {
  show.value = false;
};

onMounted(() => {
  window.addEventListener('scroll', handleScroll);
});

onUnmounted(() => {
  window.removeEventListener('scroll', handleScroll);
});

onClickOutside(DropdownMenuContainer, () => show.value = false);
```

File: packages/vue3/src/components/DropdownMenu/DropdownMenu.vue [hunk 3]
```diff
@@ -0,0 +1,85 @@
+<script setup lang="ts">
+import { nextTick, ref } from 'vue';
```
Comment (mosarabi) [inline packages/vue3/src/components/DropdownMenu/DropdownMenu.vue:2]:
```suggestion
import { nextTick, ref, onMounted, onUnmounted} from 'vue';
```

File: packages/vue3/src/components/DropdownMenu/DropdownMenu.vue (hunk 2 above)
Comment (mosarabi) [reply packages/vue3/src/components/DropdownMenu/DropdownMenu.vue:25]: would you mind to add scroll event to hide menu when user start scrolling

**Review by mosarabi — CHANGES_REQUESTED** [review]

File: packages/vue3/src/components/DropdownMenu/DropdownMenuTrigger.vue [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+  </div>
+</template>
+
+<style scoped>
+.dropdown-menu-trigger {
+  width: max-content;
+  max-width: 100%;
+  margin: 0;
+}
+</style>
```
Comment (mosarabi) [inline packages/vue3/src/components/DropdownMenu/DropdownMenuTrigger.vue:13]: I think no need to create this component 
What d u think?

**Review by mosarabi — CHANGES_REQUESTED** [review]

**Comment by eihabkhan1** [comment]
Can we set the cursor to pointer on the items

**Comment by GhitaNektt** [comment]
> Can we set the cursor to pointer on the items

yeah sure

**Pull request merged**